### PR TITLE
Closes #41 , closes #47 Cart deletion and refresh animation

### LINKED ISSUE #41: Add animation for cart refresh

### LINKED ISSUE #47: Removing line items from the cart
We will do native iOS table view removal.

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
- Added support for removing line item from the active cart
- Added cart refresh animation

### CODE REVIEW

File: Sunrise/ViewModels/CartViewModel.swift [hunk 1]
```diff
@@ -98,7 +119,22 @@ class CartViewModel: BaseViewModel {
...
+    private func deleteLineItemAtIndexPath(indexPath: NSIndexPath) {
+        if let cartId = cart.value?.id, version = cart.value?.version, lineItemId = cart.value?.lineItems?[indexPath.row].id {
+            self.isLoading.value = true
+            Commercetools.Cart.update(cartId, version: version, actions: [["action": "removeLineItem",
+                                                                           "lineItemId": lineItemId]], result: { result in
+                if let errors = result.errors where result.isFailure {
+                    super.alertMessageObserver.sendNext(self.alertMessageForErrors(errors))
+                    self.isLoading.value = false
+                }
+                self.queryForActiveCart()
```
Comment (cneijenhuis) [inline Sunrise/ViewModels/CartViewModel.swift]: Why? If the update was successful, the returned cart is the active cart.
Can't you invoke `updateCart` with the returned cart?

Comment (nikola-mladenovic) [inline Sunrise/ViewModels/CartViewModel.swift]: `self.queryForActiveCart()` also performs a call for running recalculation on the API side before retrieving the new "recalculated" instance.

Comment (cneijenhuis) [inline Sunrise/ViewModels/CartViewModel.swift]: That's a good point, but I'd prefer to add the "recalculate" as the second action to the request in line 125. Then we'd only have a single request to the API 👍

**Comment by cneijenhuis** [comment]
The animation looks great 👍

File: Sunrise/ViewModels/CartViewModel.swift (hunk 1 above)
Comment (nikola-mladenovic) [inline Sunrise/ViewModels/CartViewModel.swift]: Updated deletion logic, so now we do have a single request 👍

**Comment by cneijenhuis** [comment]
LGTM 👍

**Pull request merged**